### PR TITLE
[Conflict Fix] Partial Rust Enum Serialization/Deserialization Support 

### PR DESCRIPTION
**Pull request by calvinbrown085**
This fixes conflicts introduced here.
https://github.com/flavray/avro-rs/pull/60

All credit goes to @jdeschenes for implementing the original feature.

### CODE REVIEW

**Comment by calvinbrown085** [comment]
I'm not exactly sure what went wrong with the build here..

**Comment by calvinbrown085** [comment]
@poros Do you know what went wrong on the build? It looks like `cargo check` worked fine.

**Comment by poros** [comment]
I am not really sure what the first error is, perhaps something transient caused by travis? Anyways, I have re-started the build and it fails with compilation errors. Maybe you can try to run `make test` locally and see if your branch works on your machine?

**Comment by codehearts** [comment]
It looks like you just forgot to `use serde::Serialize;` in the test module of `de.rs`, the other two things were deprecation warnings that shouldn't break the stable and nightly builds

File: src/de.rs [hunk 1]
```diff
@@ -399,6 +559,227 @@ impl<'de> de::Deserializer<'de> for StringDeserializer {
...
 pub fn from_value<'de, D: Deserialize<'de>>(value: &'de Value) -> Result<D, Error> {
-    let mut de = Deserializer::new(value);
-    D::deserialize(&mut de)
+    let de = Deserializer::new(value);
+    D::deserialize(& de)
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
```
Comment (codehearts) [inline src/de.rs]: These tests use `serde::Serialize` but it's not defined above, so it should be defined here as

```suggestion
    use super::*;
    use serde::Serialize;
```

I tried this locally and all tests passed, so hopefully it's as simple as that! It was really sweet to see enums parsed effortlessly when I tried out this branch

**Comment by calvinbrown085** [comment]
@codehearts Thanks! I must have missed that when fixing merge conflicts

**Comment by codehearts** [comment]
Ah, now it looks like `cargo fmt` needs to be run on the repo for the build to be happy

**Comment by calvinbrown085** [comment]
@codehearts I will grab this later tonight :)

**Comment by poros** [comment]
@calvinbrown085 are you still interested in this PR? It was pretty close to land!

**Comment by calvinbrown085** [comment]
@poros Yes! I will grab this.

**Comment by calvinbrown085** [comment]
@poros Okay I ran `cargo fmt` lets see if this passes :)

**Comment by calvinbrown085** [comment]
@poros Anything else here?

**Comment by poros** [comment]
Nothing else, thanks :)

**Pull request merged**